### PR TITLE
Comments final

### PR DESCRIPTION
**Pull request by SuragNuthulapaty**
Added comments for all files in midas documenting code

### CODE REVIEW

File: MIDAS/src/data_logging.cpp [hunk 1]
```diff
@@ -2,9 +2,15 @@
 #include "log_format.h"
 #include "log_checksum.h"
 
+/**
+ * Forward decleration of the id recieving function
```
Comment (rbhog) [inline MIDAS/src/data_logging.cpp:6]: what happened to the `@brief` here and capitalized ID

File: MIDAS/src/data_logging.cpp [hunk 2]
```diff
@@ -29,6 +40,14 @@ void log_reading(LogSink& sink, Reading<T>& reading) {
     sink.write((uint8_t*) &reading.data, sizeof(T));
 }
 
+/**
+ * @brief wites a SensorData's entire queue reading to a sink
```
Comment (rbhog) [inline MIDAS/src/data_logging.cpp:44]: misspelled writes

File: MIDAS/src/data_logging.cpp [hunk 3]
```diff
@@ -63,6 +93,15 @@ void log_data(LogSink& sink, RocketData& data) {
 #ifndef SILSIM
 #define MAX_FILES 999
 
+/**
+ * @brief names a new fole for a log sink depending on the files currently on said LogSink
```
Comment (rbhog) [inline MIDAS/src/data_logging.cpp:97]: misspelled file

File: MIDAS/src/finite-state-machines/fsm.h [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
 #include "thresholds.h"
 #include "sensor_data.h"
 #include "Buffer.h"
 #include "rocket_state.h"
 
-
+/**
+ * @struct StateEstimate
+ * 
+ * @brief Holds current alttitude, acceleration, jerk, and speed values based off of averages and derivatives.
```
Comment (rbhog) [inline MIDAS/src/finite-state-machines/fsm.h:15]: misspelled altitude

File: MIDAS/src/gnc/example_kf.h [hunk 5]
```diff
@@ -2,7 +2,6 @@
 
 #include "kalman_filter.h"
 
-
 // makes a kalman filter with 9 state variables and 3 sensor inputs
 class ExampleKalmanFilter : public KalmanFilter<9, 3>
```
Comment (rbhog) [inline MIDAS/src/gnc/example_kf.h:6]: imma have to update this template at some point, doesn't really work for extended kalman filters

File: MIDAS/src/hardware/Orientation.cpp [hunk 6]
```diff
@@ -35,17 +49,38 @@ Vec3 quaternionToEuler(float qr, float qi, float qj, float qk, bool degrees) {
     return euler;
 }
 
+/**
+ * @brief Takes a rotation quaternion and turns it into its Euler 3D counterpart
```
Comment (rbhog) [inline MIDAS/src/hardware/Orientation.cpp:53]: euler angles

File: MIDAS/src/hardware/Orientation.cpp [hunk 7]
```diff
@@ -35,17 +49,38 @@ Vec3 quaternionToEuler(float qr, float qi, float qj, float qk, bool degrees) {
...
+ * 
+ * @param rotational_vector Rotation quaternion
+ * @param degrees Quaternion degrees, not used
+ * 
+ * @return Euler 3D vector representation of the quaternion
```
Comment (rbhog) [inline MIDAS/src/hardware/Orientation.cpp:58]: euler angle vector

File: MIDAS/src/hardware/Orientation.cpp [hunk 8]
```diff
@@ -35,17 +49,38 @@ Vec3 quaternionToEuler(float qr, float qi, float qj, float qk, bool degrees) {
...
 
 }
 
+/**
+ * @brief Takes a gyroscope quaternion and turns it into its Euler 3D counterpart
+ * 
+ * @param rotational_vector Gyroscope quaternion
+ * @param degrees Quaternion degrees, not used
+ * 
+ * @return Euler 3D vector representation of the quaternion
```
Comment (rbhog) [inline MIDAS/src/hardware/Orientation.cpp:72]: *euler angles*

File: MIDAS/src/hardware/telemetry_backend.cpp [hunk 9]
```diff
@@ -68,10 +76,20 @@ ErrorCode TelemetryBackend::init() {
     return ErrorCode::NoError;
 }
 
+/**
+ * @brief Gets rssi of recent packets
```
Comment (rbhog) [inline MIDAS/src/hardware/telemetry_backend.cpp:80]: capitalize rssi

File: MIDAS/src/hardware/telemetry_backend.cpp [hunk 10]
```diff
@@ -68,10 +76,20 @@ ErrorCode TelemetryBackend::init() {
...
+ * @return RSSI of most recent packet
+*/
 int8_t TelemetryBackend::getRecentRssi() {
     return rf95.lastRssi();
 }
 
+/**
+ * @brief Sets new frequency for the LoRa module
+ * 
+ * @param freq New frequence to set the LoRa module to
```
Comment (rbhog) [inline MIDAS/src/hardware/telemetry_backend.cpp:91]: frequency

File: MIDAS/src/rocket_state.h [hunk 11]
```diff
@@ -6,18 +6,35 @@
...
 template<typename SensorData>
 struct Reading {
     uint32_t timestamp_ms;
     SensorData data;
 };
 
+/**
+ * @struct SensorData 
+ * 
+ * @brief Wrapper for thread safe sensore data stroage
```
Comment (rbhog) [inline MIDAS/src/rocket_state.h:34]: storage

File: MIDAS/src/rocket_state.h [hunk 12]
```diff
@@ -27,24 +44,57 @@ struct SensorData {
 public:
     SensorData() : current(S()) { }
 
+    /**
+     * @brief pushes to the queue with new data
+     * 
+     * @param data New data of the templeted type to add to queue
```
Comment (rbhog) [inline MIDAS/src/rocket_state.h:50]: templated

File: MIDAS/src/sensor_data.h [hunk 13]
```diff
@@ -49,14 +50,25 @@ struct Acceleration {
     }
 };
 
+/**
+ * @struct euler_t
+ * 
+ * @brief euler represnetation of rotation
```
Comment (rbhog) [inline MIDAS/src/sensor_data.h:56]: representation

File: MIDAS/src/sensor_data.h [hunk 14]
```diff
@@ -76,6 +93,11 @@ struct HighGData {
     HighGData(float x, float y, float z) : ax(x), ay(y), az(z) {}
 };
 
+/**
+ * @struct LowGLSM
+ * 
+ * @brief data from the LowGLSM sensor
```
Comment (rbhog) [inline MIDAS/src/sensor_data.h:99]: Low G LSM

File: MIDAS/src/telemetry_packet.h [hunk 15]
```diff
@@ -3,6 +3,11 @@
 #include <array>
 #include <cstdint>
 
+/**
+ * @struct TelemetryPacket
+ * 
+ * @brief format of the telemetry acket
```
Comment (rbhog) [inline MIDAS/src/telemetry_packet.h:9]: packet

**Review by rbhog — CHANGES_REQUESTED** [review]

**Review by aidancostello — APPROVED** [review]

**Review by rbhog — APPROVED** [review]

**Pull request merged**